app/src/main/java/org/thoughtcrime/securesms/ApplicationContext.java

```diff
@@ -236,6 +239,9 @@ public void onCreate() {
         resubmitProfilePictureIfNeeded();
         loadEmojiSearchIndexIfNeeded();
         EmojiSource.refresh();
+
+        NetworkConstraint networkConstraint = new NetworkConstraint.Factory(this).create();
+        HTTP.INSTANCE.setConnectedToNetwork(networkConstraint::isMet);
```

Comment:
This is an interesting addition, I'm guessing it might be useful in future to prevent retrying when we don't have network?

Reply:
Mm yea - I figured since we already had the permission it didn't hurt adding it in, and it allows us to disambiguate the error a bit. Hooking it into the JobRunner and flagging which jobs require internet access could definitely help prevent "false failures" for jobs